app/src/main/java/com/babylon/wallet/android/presentation/dialogs/info/InfoDialog.kt

```diff
@@ -0,0 +1,261 @@
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalUriHandler
+import androidx.compose.ui.platform.UriHandler
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import com.babylon.wallet.android.R
+import com.babylon.wallet.android.designsystem.theme.RadixTheme
+import com.babylon.wallet.android.presentation.dialogs.info.InfoViewModel.Companion.GLOSSARY_ANCHOR
+import com.babylon.wallet.android.presentation.ui.RadixWalletPreviewTheme
+import com.babylon.wallet.android.presentation.ui.composables.BottomSheetDialogWrapper
+import com.mikepenz.markdown.compose.components.MarkdownComponent
+import com.mikepenz.markdown.compose.components.markdownComponents
+import com.mikepenz.markdown.m3.Markdown
+import com.mikepenz.markdown.m3.markdownColor
+import com.mikepenz.markdown.m3.markdownTypography
+import com.mikepenz.markdown.model.markdownAnnotator
+import com.mikepenz.markdown.model.markdownPadding
+import com.mikepenz.markdown.utils.MARKDOWN_TAG_URL
+import com.mikepenz.markdown.utils.buildMarkdownAnnotatedString
+import org.intellij.markdown.MarkdownElementTypes.INLINE_LINK
+import org.intellij.markdown.MarkdownElementTypes.LINK_DESTINATION
+import org.intellij.markdown.MarkdownElementTypes.LINK_LABEL
+import org.intellij.markdown.MarkdownElementTypes.LINK_TEXT
+import org.intellij.markdown.MarkdownTokenTypes
+import org.intellij.markdown.ast.ASTNode
+import org.intellij.markdown.ast.findChildOfType
+import org.intellij.markdown.ast.getTextInNode
+
+typealias DSR = com.babylon.wallet.android.designsystem.R.drawable
+
+@Composable
+fun InfoDialog(
+    modifier: Modifier = Modifier,
+    viewModel: InfoViewModel,
+    onDismiss: () -> Unit
+) {
+    val glossaryItem by viewModel.glossaryItem.collectAsStateWithLifecycle()
+
+    BottomSheetDialogWrapper(
+        modifier = modifier,
+        showDragHandle = true,
+        dragToDismissEnabled = true,
+        onDismiss = onDismiss
+    ) {
+        InfoDialogContent(
+            scrollState = ScrollState(initial = 0),
```

Comment:
Maybe `rememberScrollState`? I think this will also fix the issue of the scroll being reset (scrolled up) after clicking a URL and getting back to the app (or after pausing and resuming the app while the info pop up is displayed)

Reply:
I did this on purpose for the glossary items. For example, when you click on `XRD` it will scroll to the top so you can see the content from the beginning.

Reply:
Makes sense, then to improve this and also fix the issue I described maybe it's worth using `rememberScrollState` and scrolling to the top based on a boolean (using LaunchedEffect) only when tapping on a glossary item and not a URL.

Reply:
did ti with a slightly different [way](https://github.com/radixdlt/babylon-wallet-android/pull/1169/commits/35c58ac69f1ea97d806ebe042d5a09bc0f318718)

Reply:
Even better!